src/ast/irbuilderbpf.cpp

```diff
@@ -514,10 +514,10 @@ Value *IRBuilderBPF::createScratchBuffer(
   auto max = CreateLoad(getInt64Ty(),
                         module_.getGlobalVariable(to_string(
                             bpftrace::globalvars::GlobalVar::MAX_CPU_ID)));
-  // Verify CPU ID is between 0 and MAX_CPU_ID to ensure we pass BPF
-  // verifer on older 5.2 kernels
-  auto cmp = CreateICmp(CmpInst::ICMP_ULE, cpu_id, max, "cpuid.min.cmp");
-  auto select = CreateSelect(cmp, cpu_id, max, "cpuid.min.select");
+  // Mask CPU ID by MAX_CPU_ID to ensure BPF verifier knows CPU ID is bounded
+  // on older kernels. Note this means MAX_CPU_ID must be N^2 - 1 for some N.
```

Comment:
You mean this right?
```suggestion
  // on older kernels. Note this means MAX_CPU_ID must be 2^N - 1 for some N.
```

Reply:
Yes, math is hardz